src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/TextMapPropagator.cs

```diff
@@ -0,0 +1,139 @@
+
+        /// <summary>
+        /// Extracts the value from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation trace Id and trace state from carrier.</param>
+        /// <param name="traceId">The extracted trace Id from the carrier.</param>
+        /// <param name="traceState">The extracted trace state from the carrier.</param>
+        public abstract void ExtractTraceIdAndState(object carrier, PropagatorGetterCallback getter, out string? traceId, out string? traceState);
+
+        /// <summary>
+        /// Extracts the baggage key-value pairs list from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation baggage list from carrier.</param>
+        /// <returns>Returns the extracted key-value pair list from teh carrier.</returns>
+        public abstract IEnumerable<KeyValuePair<string, string?>>? ExtractBaggage(object carrier, PropagatorGetterCallback getter);
+
+        /// <summary>
+        /// Get or set the process wide propagator object which used as the current selected propagator.
+        /// </summary>
+        public static TextMapPropagator Current
+        {
+            get
+            {
+                Debug.Assert(s_current is not null);
+                return s_current;
+            }
+
+            set
+            {
+                if (value is not null)
+                {
+                    s_current = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// returns the default propagator object which Current property will be initialized with.
+        /// </summary>
+        public static TextMapPropagator CreateDefaultPropagator() => LegacyPropagator.Instance;
+
+        /// <summary>
+        /// returns the propagator which can propagate the trace context data using the root Activity and ignore any created child activities.
+        /// </summary>
+        public static TextMapPropagator CreatePassThroughPropagator() => PassThroughPropagator.Instance;
+
+        /// <summary>
+        /// returns the propagator which suppress injecting any data to teh carriers.
+        /// </summary>
+        public static TextMapPropagator CreateNoOutputPropagator() => NoOutputPropagator.Instance;
+
+        // internal stuff
+
+        internal static void InjectBaggage(object carrier, IEnumerable<KeyValuePair<string, string?>> baggage, PropagatorSetterCallback setter, bool injectAsW3C = false)
```

Comment:
```suggestion
        internal static void InjectBaggage(object carrier, IEnumerable<KeyValuePair<string, string?>> baggage, PropagatorSetterCallback setter)
```
I think the w3c path is dead code so we can eliminate it?

Reply:
I think you reviewed a pervious revision as I already eliminated the W3C path before. Anyway, we should be good here.